forge/src/main/java/org/openjdk/skara/forge/gitlab/GitLabMergeRequest.java

```diff
@@ -710,6 +698,12 @@ public void setLabels(List<String> labels) {
 
     @Override
     public List<Label> labels() {
+        if (labels == null) {
+            labels = request.get("").execute().get("labels").stream()
+                    .map(JSONValue::asString)
+                    .sorted()
+                    .collect(Collectors.toList());
+        }
```

Comment:
I don't know if we need to make it consistent with GitHubPullRequest but I did.

Reply:
Please note that GitLabMergeRequest handles labels a bit differently to JiraIssue and GitHubPullRequest. The internal cache is a list of String, and we override the labelNames() method. This is all because of an optimization I made recently.

This means that the lazy fetch needs to happen in labelNames() instead, and the labels() method should call labelNames() to get the list to convert to Label objects.